reconcile/utils/gql.py

```diff
@@ -86,16 +88,20 @@ def __init__(self, url, token=None, int_name=None, validate_schemas=False):
         self.token = token
         self.integration = int_name
         self.validate_schemas = validate_schemas
-        self.client = GraphQLClient(self.url)
+        self._req_headers = None
```

Comment:
I'd put a `None` return type hint on `__init__` so that Mypy will type check this code. I suspect that when you do so, it will probably be unhappy that `_req_headers` starts as `None`, but conditionally becomes a `dict`. The solution would be to type it as an `Optional[dict[str, str]]`.

Reply:
I added `None` return type hint. Also used DI to inject `Client` for better testability.